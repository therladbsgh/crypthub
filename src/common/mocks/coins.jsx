export const coins = [
    {
        name: 'Bitcoin',
        symbol: 'BTC',
        currPrice: 10000
    },
    {
        name: 'Ethereum',
        symbol: 'ETH',
        currPrice: 5000
    }
]
