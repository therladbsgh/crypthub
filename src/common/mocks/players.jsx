import date from 'date-and-time';

export const player1 = {
    name: 'player1',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 250
        },
        {
            name: 'Ethereum',
            amount: 1000.28373623
        },
        {
            name: 'USD',
            amount: 250
        }
    ],
    transactions: {
        history: [
            {
                id: '1',
                type: 'market',
                side: 'buy',
                size: 25,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 1 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: true,
                filledDate: date.parse('Apr 3 2018', 'MMM D YYYY')
            },
            {
                id: '2',
                type: 'limit',
                side: 'buy',
                size: 35.000006,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('Apr 5 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ],
        current: [
            {
                id: '3',
                type: 'short',
                side: 'sell',
                size: 80,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: false,
                filledDate: null
            },
            {
                id: '4',
                type: 'limit',
                side: 'buy',
                size: 2,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 7 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('May 2 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ]
    },
    netWorth: 200,
    numTrades: 80,
    netReturn: -20,
    todayReturn: 5,
    currRank: 2,
    buyingPower: 150,
    shortReserve: 50
};

export const player2 = {
    name: 'player2',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 300
        }
    ],
    transactions: {
        history: [
            {
                id: '5',
                side: 'buy',
                size: 1,
                price: 10,
                symbol: 'ETH'
            },
            {
                id: '6',
                side: 'sell',
                size: 3500,
                price: 202220,
                symbol: 'ETH'
            }
        ],
        current: [
            {
                id: '7',
                side: 'sell',
                size: 23230,
                price: 232,
                symbol: 'BTC'
            },
            {
                id: '8',
                side: 'buy',
                size: 6000,
                price: 50,
                symbol: 'ETH'
            }
        ]
    },
    netWorth: 300,
    numTrades: 2,
    netReturn: 120
};
