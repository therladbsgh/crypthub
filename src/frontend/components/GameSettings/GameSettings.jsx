import * as _ from 'lodash';
import React, { Component } from 'react';
import formatCurrency from 'format-currency';
import { Header, Button, Form, Message, Grid } from 'semantic-ui-react';
import { GameBackend } from 'endpoints';
import { GameCard, JoinModal } from 'components';
import { GameSettingsStyle as styles } from 'styles';

export default class GameSettings extends Component {
    constructor(props) {
        super(props);

        this.state = {
            gameObj: {
                ...this.props.game,
                password: '',
                passwordConfirm: ''
            },
            submitted: false,
            loading: false,            
            errMsg: '',
            errField: ''
        };

        this.handleLeave = this.handleLeave.bind(this);
        this.handleInvite = this.handleInvite.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.setAccess = this.setAccess.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleLeave() {
        //TODO: logic
    }

    handleInvite() {
        //TODO: logic
    }

    setAccess(bool) {
		const { gameObj } = this.state;
		const { password, passwordConfirm } = gameObj;

		this.setState({
			gameObj: {
				...this.state.gameObj,
				isPrivate: bool,
				password: bool ? password : '',
				passwordConfirm: bool ? passwordConfirm : '',
			}
		});
    }
    
    handleChange(event) {
		const { gameObj } = this.state;
		const target = event.target;
		const value = target.value;

		if (_.has(gameObj, target.name)) {
			this.setState({
				gameObj: {
					...gameObj,
					[target.name]: value
				}
			});
		} else {
			this.setState({
				[target.name]: value
			});
		}
	}

    handleSubmit() {
        const { gameObj } = this.state;
        const { isPrivate, password, passwordConfirm } = gameObj;

        event.preventDefault();
        this.setState({
            errMsg: '',
            errField: '',
            submitted: false,
            loading: true
        });

        if (isPrivate && password !== passwordConfirm) {
            return this.setState({
                loading: false,
                errMsg: 'Passwords don\'t match.',
                errField: 'password'
            });
		}

		const gameObjSend = _.omit(gameObj, ['passwordConfirm']);

        GameBackend.saveGame(gameObjSend)
		.then(res => {
			console.log('success! ', res);
			this.setState({
				submitted: true,
				loading: false
			});
		}, ({ err, field }) => {
			console.log('error! ', err);
			this.setState({
                submitted: true,                
                loading: false, 
				errMsg: err,
				errField: field
			});
		});
    }

    render() {
        const { inGame, isHost, userId } = this.props;
        const { gameObj, submitted, loading, errMsg, errField } = this.state;
        const { id, host, playerPortfolioPublic, startingBalance, commissionValue, shortSelling, limitOrders, stopOrders, isPrivate, password, passwordConfirm, completed } = gameObj;

        return (
			<div>
                {!completed &&
                [<Header key='1' as='h2'>Game Actions</Header>,
                isHost && <Button key='2' onClick={this.handleInvite} positive>Invite Players</Button>,
                inGame ?
                <Button key='3' onClick={this.handleLeave} negative>Leave Game</Button>
                :
                <JoinModal key='3' size='medium' isPrivate={isPrivate} gameId={id} userId={userId} />]}
				<Header as='h2'>About This Game</Header>
                <GameCard game={this.props.game} />
                {isHost && !completed &&
                [<Header key='header' as='h2'>Game Options</Header>,
                <Form key='form' loading={loading} success={submitted && !errMsg} error={!!errMsg}>
                    <Form.Field>
                        <label>Game Access</label>
                        <Button.Group>
                            <Button color={isPrivate ? null : 'blue'} onClick={() => this.setAccess(false)}>Public</Button>
                            <Button color={isPrivate ? 'blue' : null} onClick={() => this.setAccess(true)}>Private</Button>
                        </Button.Group>
                    </Form.Field>
                    {isPrivate && 
                    <Form.Group widths='equal'>
                        <Form.Field error={errField === 'password'}>
                            <label>Password</label>
                            <input type='password' placeholder='Password' name='password' value={password} onChange={this.handleChange} />
                        </Form.Field>
                        <Form.Field error={errField === 'password'}>
                            <label>Confirm Password</label>
                            <input type='password' placeholder='Confirm password' name='passwordConfirm' value={passwordConfirm} onChange={this.handleChange} />
                        </Form.Field>
                    </Form.Group>}
                    <Message
						success
						header='Settings Change Successful!'
						content="The game settings have been saved."
					/>
                    <Message
                        error
                        header='Error'
                        content={errMsg}
                    />
                    <Button onClick={this.handleSubmit} positive>Save Changes</Button>
                </Form>]}
                <Header as='h2'>Game Settings</Header>
                <Grid className={styles.settings} columns={2}>
                    <Grid.Row>
                        <Grid.Column>
                            <div className={`${styles.column} ${styles.firstColumn}`}>
                                <label>Player Portfolios:</label>
                                <p>{playerPortfolioPublic ? 'Public' : 'Private'}</p>
                            </div>
                        </Grid.Column>
                        <Grid.Column>
                            <div className={`${styles.column} ${styles.firstColumn}`}>
                                <label>Starting Balance:</label>
                                <p>{formatCurrency(startingBalance, { format: '%s%v', symbol: '$' })}</p>
                            </div>
                        </Grid.Column>
                    </Grid.Row>
                    <Grid.Row className={styles.row}>
                        <Grid.Column>
                            <div className={styles.column}>
                                <label>Trade Commission:</label>
                                <p>{formatCurrency(commissionValue, { format: '%s%v', symbol: '$' })}</p>
                            </div>
                        </Grid.Column>
                        <Grid.Column>
                            <div className={styles.column}>
                                <label>Short Selling:</label>
                                <p>{shortSelling ? 'Enabled' : 'Disabled'}</p>
                            </div>
                        </Grid.Column>
                    </Grid.Row>
                    <Grid.Row className={styles.row}>
                        <Grid.Column>
                            <div className={styles.column}>
                                <label>Limit Orders:</label>
                                <p>{limitOrders ? 'Enabled' : 'Disabled'}</p>
                            </div>
                        </Grid.Column>
                        <Grid.Column>
                            <div className={styles.column}>
                                <label>Stop Orders:</label>
                                <p>{stopOrders ? 'Enabled' : 'Disabled'}</p>
                            </div>
                        </Grid.Column>
                    </Grid.Row>
                    {(!isHost || completed) &&
                    <Grid.Row className={styles.row}>
                        <Grid.Column>
                            <div className={styles.column}>
                                <label>Game Access:</label>
                                <p>{isPrivate ? 'Private' : 'Public'}</p>
                            </div>
                        </Grid.Column>
                    </Grid.Row>}
                </Grid>
			</div>
        );
    }
}
