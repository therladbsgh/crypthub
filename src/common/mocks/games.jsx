import date from 'date-and-time';
import * as PlayerMocks from './players';

export const game1 = {
    id: '1',
    name: 'game1',
    description: 'desc1',
    host: 'user1',
    created: date.parse('Apr 1 2018', 'MMM D YYYY'),
    start: date.parse('Apr 2 2018', 'MMM D YYYY'),
    end: date.parse('May 7 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game2 = {
    id: '2',
    name: 'game2',
    description: 'desc2',
    host: 'user2',
    created: date.parse('Apr 2 2018', 'MMM D YYYY'),
    start: date.parse('Apr 3 2018', 'MMM D YYYY'),
    end: date.parse('May 1 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game3 = {
    id: '3',
    name: 'game3',
    description: 'desc3',
    host: 'user3',
    created: date.parse('Apr 4 2018', 'MMM D YYYY'),
    start: date.parse('Apr 8 2018', 'MMM D YYYY'),
    end: date.parse('Apr 28 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};
