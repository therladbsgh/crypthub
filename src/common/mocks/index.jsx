import * as GameMocks from './games';
import * as PlayerMocks from './players';

export { GameMocks, PlayerMocks };
