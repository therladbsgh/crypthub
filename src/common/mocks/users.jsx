import * as GameMocks from './games';

export const user1 = {
    id: 'id1',
    name: 'user1',
    ELO: 3000,
    games: [
        GameMocks.game1,
        GameMocks.game2
    ]
};

export const user2 = {
    id: 'id2',
    name: 'user2',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};

export const user3 = {
    id: 'id3',
    name: 'user3',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};
