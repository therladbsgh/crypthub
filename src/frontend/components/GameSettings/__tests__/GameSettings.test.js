import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import GameSettings from '../GameSettings';
import { GameMocks } from 'mocks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<Router><GameSettings game={GameMocks.game1} /></Router>).toJSON();
  	expect(tree).toMatchSnapshot();
});