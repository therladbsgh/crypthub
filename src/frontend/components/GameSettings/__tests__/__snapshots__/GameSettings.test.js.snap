// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    className="ui header"
  >
    Game Actions
  </h2>
  <button
    className="ui medium primary button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    <i
      aria-hidden="true"
      className="user add icon"
    />
    Join Game
  </button>
  <h2
    className="ui header"
  >
    About This Game
  </h2>
  <div
    className="ui card"
    href={undefined}
    id={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        <a
          href="/game/1"
          onClick={[Function]}
        >
          game1
        </a>
        <span
          className="false undefined"
        >
          
        </span>
      </div>
      <div
        className="meta"
      >
        desc1
      </div>
      <div
        className="description"
      >
        Created 
        <strong>
          37 days
           ago
        </strong>
         by 
        <strong>
          user1
        </strong>
         to start on 
        <strong>
          04/02/2018
        </strong>
         and end 
        on
         
        <strong>
          05/07/2018
        </strong>
      </div>
    </div>
    <div
      className="extra content"
    >
      2
       player
      s
    </div>
  </div>
  <h2
    className="ui header"
  >
    Game Settings
  </h2>
  <div
    className="ui two column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="undefined undefined"
        >
          <label>
            Player Portfolios:
          </label>
          <p>
            Public
          </p>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="undefined undefined"
        >
          <label>
            Starting Balance:
          </label>
          <p>
            $10,000.00
          </p>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className={undefined}
        >
          <label>
            Trade Commission:
          </label>
          <p>
            $10.00
          </p>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className={undefined}
        >
          <label>
            Short Selling:
          </label>
          <p>
            Enabled
          </p>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className={undefined}
        >
          <label>
            Limit Orders:
          </label>
          <p>
            Enabled
          </p>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className={undefined}
        >
          <label>
            Stop Orders:
          </label>
          <p>
            Enabled
          </p>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className={undefined}
        >
          <label>
            Game Access:
          </label>
          <p>
            Public
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
